manifests/server.pp

```diff
@@ -36,19 +36,33 @@
     },
     File { '/etc/puppetlabs/pds/ssl/ca.pem':
       ensure => file,
-      source => "/etc/puppetlabs/puppet/ssl/certs/ca.pem",
+      source => '/etc/puppetlabs/puppet/ssl/certs/ca.pem',
     },
   ]
 
+  if $database_host != undef {
+    $db_host = $database_host
+  } else {
+    # query PuppetDB for the database host
+    $query_output = puppetdb_query('nodes[certname] { resources{type="Class" and title="Puppet_enterprise::Profile::Database"}}')
```

Comment:
```suggestion
    $query_output = puppetdb_query('resources[certname] {type="Class" and title="Puppet_enterprise::Profile::Certificate_authority"}')
```

I think we decided to default to using the primary as the database server, if not specified. This should uniquely identify the primary server.

If users are using more advanced architectures, it may be wrong. They will need to define the DB server themselves, in that situation, and that's okay.

Reply:
@reidmv I don't remember deciding that and the ticket is a bit ambiguous.
If we configure the primary as the db server by default, the XL architecture will not work.
If we configure the database server by default, by choosing the 1st returned database server, will the XL configuration work properly? If so, that seems a more robust solution to me.

Reply:
@timidri if a default selection isn't predictable and obvious, then it's not (IMO) an acceptable default selection. The problem with picking the "first" item in a list that doesn't have a guaranteed ordering is that it's effectively random. And, if something happens to change that list order at some point, we could actively trigger a cascading misconfiguration.

The primary is non-ideal in the XL architecture, but it will work. I will assert that defaults need to cover the "standard" architecture, but by the time we're talking about XL it's appropriate for a user to step in and choose the DB themselves.

It's potentially arguable that even in XL, the right place for PDS data is the primary. That the separate PostgreSQL should continue to be reserved for the PuppetDB workload only.

Reply:
@reidmv ok, I'll change the query to look up the primary.